### PR TITLE
[WIP] Symfony 2.8/3.0 support

### PR DESCRIPTION
**Pull request by XWB**

### CODE REVIEW

**Comment by stof** [comment]
injecting the request as a service has also been deprecated, so this is not good.

Use the RequestStack instead (and keep the current logic with the active scope check when running on 2.3 which does not have the RequestStack)

**Comment by XWB** [comment]
@stof I've just made some changes.

File: EventListener/ThreadPermalinkListener.php [hunk 1]
```diff
@@ -29,6 +30,11 @@ class ThreadPermalinkListener implements EventSubscriberInterface
     protected $container;
 
     /**
+     * @var RequestStack
+     */
+    protected $requestStack;
```
Comment (stof) [inline EventListener/ThreadPermalinkListener.php]: please make it private

File: EventListener/ThreadPermalinkListener.php [hunk 2]
```diff
@@ -45,12 +51,26 @@ public function __construct(ContainerInterface $container)
...
-        if (!$this->container->isScopeActive('request')) {
-            return;
+        if (null === $this->requestStack) {
+            if (!$this->container->isScopeActive('request')) {
+                return;
+            }
+
+            $request = $this->container->get('request');
+        } else {
+            $request = $this->requestStack->getCurrentRequest();
```
Comment (stof) [inline EventListener/ThreadPermalinkListener.php]: you need to account for ``$request`` being null here too

File: Resources/config/events.xml [hunk 3]
```diff
@@ -35,6 +35,9 @@
         <service id="fos_comment.listener.thread_permalink" class="%fos_comment.listener.thread_permalink.class%">
             <argument type="service" id="service_container" />
             <tag name="kernel.event_subscriber" />
+            <call method="setRequest">
```
Comment (stof) [inline Resources/config/events.xml]: this is wrong

File: EventListener/ThreadPermalinkListener.php [hunk 4]
```diff
@@ -45,12 +51,26 @@ public function __construct(ContainerInterface $container)
...
 
         $thread = $event->getThread();
-        $thread->setPermalink($this->container->get('request')->getUri());
+        $thread->setPermalink($request->getUri());
+    }
+
+    /**
+     * @param RequestStack|null $requestStack
+     */
+    public function setRequestStack(RequestStack $requestStack = null)
```
Comment (stof) [inline EventListener/ThreadPermalinkListener.php]: please inject it in the constructor instead

**Comment by XWB** [comment]
@stof Done.

File: Tests/Functional/WebTestCase.php [hunk 5]
```diff
@@ -34,7 +34,7 @@ class WebTestCase extends BaseWebTestCase
     public static function assertRedirect($response, $location)
     {
         self::assertTrue($response->isRedirect(), 'Response should be a redirect, got status code: '.substr($response, 0, 2000));
-        self::assertEquals('http://localhost'.$location, $response->headers->get('Location'));
```
Comment (Ener-Getick) [inline Tests/Functional/WebTestCase.php]: Why did you remove this?

**Comment by Ener-Getick** [comment]
ping

**Comment by XWB** [comment]
Fixed via https://github.com/FriendsOfSymfony/FOSCommentBundle/pull/512
